### PR TITLE
Built to last landing page

### LINKED ISSUE #1359: Add "Built to last" landing page
Design: https://share.goabstract.com/b5bce38a-eb0e-4a1e-bd8c-bd6edc180600 

Component specs: https://www.notion.so/simplabs/Design-system-strategy-a37f5aedbcb04eff8acb32b35566a572

Content: https://www.notion.so/simplabs/Create-Built-to-Last-landing-page-175130c47fbc4f76991d9d81df3306c5

### Hero video banner

The hero has a background video that plays automatically in a loop. It is not yet certain if the hero will have a blue overlay on it or not, please use a placeholder video for now.

### Media cards

**Different kinds of media cards:** 

1. Single - default card used throughout, in particular when the image is a person's portrait photo.
2. Double-width - a loud variant. Used for information with more prominence and/or if there is an item missing (ensures the grid always looks full) 
3. Half-height - a quiet variant. Used for related blog posts, particularly when the image is a graphic rather than a person's portrait photo.

**Different kinds of supported media:** 

1. In-page video - allow the user to view videos without navigating to a different page. The video is a youtube embed, which displays controls (including full-screen mode). Allowing the user to see the video on this page helps reduce the number of pages that need to be built. 
    - On hover:
        - Image: transform the image to scale 1.03, display the play button, use a transition of 0.25s ease.
        - Text: add blue underline style that is currently used for headline links throughout the website
    - On click:
        - Image: replace with video embed with a dark background. The embed has all the usual video player UI including full-screen mode.
2. Blog post link - link to a blog post within the website. It uses the illustration that is in the blog post as a preview. Older blog posts use the visual asset provided as a fallback. For the Built to Last landing page, blog posts are selected manually. 
    - Image: transform the image to scale 1.03, use a transition of 0.25s ease.
    - On hover:
        - Text: add blue underline style that is currently used for headline links throughout the website
    - On click: send the user to the linked blog post.

## Workshop cards

**Different kinds of workshop cards:** 

1. 1-person workshop: Only one person leads the workshop
2. 2-person workshop: Two people are leading the workshop

Workshop cards: 

- On hover:
    - Background card hover: background turns darker blue, use a transition of 0.25s ease.
    - CTA hover:  Use the same hover state that is used throughout the website (link underline and movement of arrow)
- On click: send user to workshop detail page

### LINKED ISSUE #1360: Add workshop detail page
The workshop detail page gives a detailed description for the user, displaying the outline and any other relevant information. 

Design: https://share.goabstract.com/b5bce38a-eb0e-4a1e-bd8c-bd6edc180600 

Please note: 
- This layout is based on the blog post layout (uses same header & typography styles) 
- After the page title and lead, there is a rich text block (length is variable depending on the content) 
- After the rich text block, there is an unordered list with icons ("workshop overview"). Icons vary depending on the content. Number of items in the list varies depending on the content. 
- After the workshop overview, there is another rich text block (length is variable depending on the content)

### PR DESCRIPTION
**Pull request by marcoow**
This adds the built-to-last landing page(s).

| Desktop | Mobile |
|---|-----|
| desktop | mobile |

### TODO

- [x] add linking to media cards
- [x] add hover effect for video cards
- [x] generate pages dynamically
  - [x] videos: add correct images
  - [x] videos: add correct teasers (quotes from the interviews?)
  - [x] videos: fix `og:image` for @marcoow's talk an digital product development
  - [x] videos: add descriptions
  - [x] videos: ignore `og:image`s in service worker
  - [x] videos: add durations
  - [x] workshops: ignore `og:image`s in service worker
  - [x] workshops: add descriptions
  - [x] workshops: add `og:image`s
  - [x] workshops: add correct teasers
  - [x] workshops: add interface inventory & design sprint workshops
  - [x] workshops: allow SVGs for tags
  - [x] workshops: add proper icons for topic lists
  - [x] landing-page: replace semi-static content on top and bottom with real content
  - [x] landing-page: add description meta tag
  - [x] landing-page: link "become a guest" to Google questionnaire
  - [x] configure redirects in Netlify from Built to Last blog posts to new video pages
  - [x] configure redirects in Netlify from old Ember workshop routes to new workshop page
  - [x] landing page: allow which workshops are shown in which order (best thing to do is probably to create a workshop card per workshop, e.g. `WorkshopCardDigitalProductStrategy`)
- [ ] review all of the text
  - [x] are we saying what we want to say? has what we want to say changed and require changes to the text? etc.
  - [x] check for spelling and grammar errors
  - [ ] check potential for improving with regards to SEO (should we be including certain keywords, etc.?)

closes #1359
closes #1360

The scrolling bug that was discovered but that is actually unrelated to this is fixed via #1492.

**Once this is merged we should set up Google Adwords campaigns**

### CODE REVIEW

**Comment by marcoow** [comment]
@marhigh this isn't don yet (only main landing page exists, all static data etc.) but could you have a look already at `/built-to-last/` (not linked from anywhere yet) to validate before I continue?

**Comment by marhigh** [comment]
This is looking great! Good progress 🚀
- Shall we have the background video in a loop? 
- I see home variations in font sizes & spacing. How shall I provide feedback, would you like me to screenshot the developed page and annotate it?

**Comment by marcoow** [comment]
Yes, that would be helpful. All sizes and spacing should be using what we have on the rest of the site.

**Comment by marhigh** [comment]
> Yes, that would be helpful. All sizes and spacing should be using what we have on the rest of the site.

Of course. I do want to create a simplified spacing system when the time is right. For now, would you be able to send me the current spacings on the codebase for future reference? 

Here are a few comments/considerations: https://share.goabstract.com/8be8c78b-6fa7-4f58-942f-e0272de56adc

**Comment by marcoow** [comment]
I think the main aspects of the spacing system are

* the layout with the grid columns: https://github.com/simplabs/simplabs.github.io/blob/master/src/ui/styles/blocks/layout.block.css#L5
* the offset utility classes: https://github.com/simplabs/simplabs.github.io/blob/master/src/ui/styles/blocks/offset.block.css

I don't think we have that defined in a more approachable form anywhere…

**Comment by marcoow** [comment]
@marhigh this should be all done now (with fake content). Can you re-check

* workshop page: https://deploy-preview-1430--objective-northcutt-37494c.netlify.app/workshop/
* landing page: https://deploy-preview-1430--objective-northcutt-37494c.netlify.app/resources/
* video page: https://deploy-preview-1430--objective-northcutt-37494c.netlify.app/resources/video/

If everything is visually correct with the fake content, the next step is to generate the pages dynamically with our real content.

**Comment by marhigh** [comment]
@marcoow Design feedback: 

**landing page:** https://deploy-preview-1430--objective-northcutt-37494c.netlify.app/resources/
- [x] [Not in the original design] For the hero video background, let's add a subtle overlay to help the text be more readable. We can use `rgba(0,0,0,0.2)`.
- [x] Re: placement of the video in mobile, I am unsure how to move it around in inspect mode to test different positions. Could you please advise?
- [x] For the workshop cards, one of the annotations was not applied. I made a comment about it on Abstract (https://share.goabstract.com/8be8c78b-6fa7-4f58-942f-e0272de56adc?collectionLayerId=a44031d2-045d-4f64-9a69-2e33746904fe&mode=design)
- [x] The duration chip on video cards: the number is not vertically centered on the chip (it is too high)
- [x] On mobile, the spacing between the cards under "founder insights" is inconsistent

***

**workshop page:** https://deploy-preview-1430--objective-northcutt-37494c.netlify.app/workshop/
- [x] List topic item: the top of the icon and the top of the headline should be aligned (visually, which means not counting the space included in line-height)
- [x] List topic item: let's increase the spacing between items
- [x] Blue divider with the small headline: the spacing around the headline needs to be increased
- [x] The workshop leads area has quite different text sizes and spacings and colors, see them side by side: 

| Screen Shot 2021-02-09 at 11 15 43 | implementation |
|----------------------------------|----------------------------------------------------|
| Original design | Current implementation |

- [x] CTA (free workshop signup): the button does not display a pointer cursor on hover. 
- [x] CTA (free workshop signup): input field placeholder color and spacing should be updated

***

**video page:** https://deploy-preview-1430--objective-northcutt-37494c.netlify.app/resources/video/
- [x] The spacing between the media cards should be increased - this can be a global update in the resources page as well

***

**Footer**
- Are we also expecting to launch this with the updated footer that includes the newsletter signup and a promotional marking on the Playbook?
- The footer has a weird underline on nav header links that is not displayed in the live website (see screenshot) 
Screen Shot 2021-02-09 at 10 25 25

**Comment by marcoow** [comment]
> Are we also expecting to launch this with the updated footer that includes the newsletter signup and a promotional marking on the Playbook?

I would do that in a separate PR

**Comment by marcoow** [comment]
> Re: placement of the video in mobile, I am unsure how to move it around in inspect mode to test different positions. Could you please advise?

I think it's easiest if we look at that together and I screen-share.

**Comment by marcoow** [comment]
> The footer has a weird underline on nav header links that is not displayed in the live website (see screenshot)

This exists on the production site as well – moved into a separate issue: #1451

**Comment by marcoow** [comment]
> [Not in the original design] For the hero video background, let's add a subtle overlay to help the text be more readable. We can use rgba(0,0,0,0.2).

I could't quite figure out how to do this properly. The best option might be to just ask Alex to build this into the video directly.

**Comment by marcoow** [comment]
@marhigh can you re-check the ones I checked off above? I'm reusing most of the styles for spacing, text sizes etc. we have already so it might not all be 100% accurate according to your designs but close hopefully. Let me know if we need to deviate from the styles we're using elsewhere on the page anywhere.

**Comment by marhigh** [comment]
> > [Not in the original design] For the hero video background, let's add a subtle overlay to help the text be more readable. We can use rgba(0,0,0,0.2).
> 
> I could't quite figure out how to do this properly. The best option might be to just ask Alex to build this into the video directly.

Why not use a pseudo element? https://stackoverflow.com/questions/16823636/overlaying-a-div-on-top-of-html-5-video

**Comment by marcoow** [comment]
I couldn't get that to be positioned correctly

**Comment by marhigh** [comment]
> I couldn't get that to be positioned correctly

I could suggest asking for help from our team, perhaps @pichfl or @nickschot might have some insight they can share.

It would be beneficial to be able to create css overlays on our end for both this and future components.

**Comment by nickschot** [comment]
What's probably causing some trouble there is the `object-fit: contain` on the video caused by default browser styles, this (while keeping aspect-ratio) makes sure the video fits within the containing div and seems to cause some empty space above/below the video.

**Comment by marcoow** [comment]
exactly – I'd say we simply encode the effect we want directly in the video if we don't **also** need the video as it is without the effect

**Comment by marhigh** [comment]
Alright - I hope we can work on this in the future if another overlay is designed & needed. These types of components are relatively popular in web design and to get a full stop on the creation of them is quite disheartening. Up to you on how to proceed. 

The rest of the checked items look good.

**Comment by marcoow** [comment]
@marhigh background video should appear with dark overlay now

**Review by marhigh — APPROVED** [review]

**Comment by marcoow** [comment]
@marhigh left design/content feedback [here](https://www.notion.so/simplabs/Content-design-feeback-59fb80b736064d759934fe26637a5227)

**Comment by marhigh** [comment]
We're very close! Here is some final feedback: 

- [x] What made you change the hero video so that it is so short now? It seems to have changed size quite a bit, can we revert the height to what it was originally? 
- [x] The subheadline is missing a line break (before "We would love to hear from you")

  Screen Shot 2021-04-21 at 11 05 48

- [x] Workshop cards are missing a clickable background with a hover state (https://share.goabstract.com/3a3e1aa4-d8c5-43de-aecd-a2dfbfba71e4?collectionLayerId=c345895d-9d4e-4deb-ac50-057518f6d4f9&mode=design)

**Comment by marcoow** [comment]
> What made you change the hero video so that it is so short now? It seems to have changed size quite a bit, can we revert the height to what it was originally?

It didn't change recently but the height depends on text size as it's in `rem` and the text size changes with the viewport width. I made it so that it's always ca. 2/3 of the viewport height on desktop now.

**Comment by marcoow** [comment]
@marhigh should all be fixed now; also added the image-scaling-on-hover for the media cards that I missed in the issue

**Comment by marcoow** [comment]
>> What made you change the hero video so that it is so short now? It seems to have changed size quite a bit, can we revert the height to what it was originally?

> It didn't change recently but the height depends on text size as it's in rem and the text size changes with the viewport width. I made it so that it's always ca. 2/3 of the viewport height on desktop now.

This turned out to not really work in all scenarios (e.g. relatively short browser window) – I changed it again to now be consistent across screen sizes but not actually be dependent on screen size.

**Comment by marhigh** [comment]
> @marhigh should all be fixed now; also added the image-scaling-on-hover for the media cards that I missed in the issue

Hi! It seems we may have had a miscommunication with this one, as the image is supposed to scale without scaling the entire bounding box as well. 

An example can be seen here (disregard the animation of the bottom border):

https://user-images.githubusercontent.com/8226112/115698444-e7720b00-a364-11eb-8dd1-281a233ce907.mov

I will take a look at the video hero a bit later today and propose an update.

**Comment by marcoow** [comment]
ok, understand – updated the hover effect accordingly.

File: lib/generate-resources/lib/templates/landing-page/stylesheet.hbs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+@block header-content from "../HeaderContent/stylesheet.css";
+@block media-card from "../../styles/blocks/media-card.block.css";
+
+:scope {
+  block-name: {{componentName}};
+  composes: 'layout';
+}
+
+.video-header {
+  height: 62rem;
```
Comment (marhigh) [inline lib/generate-resources/lib/templates/landing-page/stylesheet.hbs]:
```suggestion
  height: 80vh;
  max-height: 70rem;
  min-height: 90rem;
```

File: lib/generate-resources/lib/templates/landing-page/stylesheet.hbs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
+  z-index: 1;
+}
+
+.video {
+  width: 100%;
+  height: 62rem;
```
Comment (marhigh) [inline lib/generate-resources/lib/templates/landing-page/stylesheet.hbs]:
```suggestion
  height: 80vh;
  max-height: 70rem;
  min-height: 90rem;
```

**Review by marhigh — COMMENTED** [review]
Here's my proposal for the height of the hero banner, let me know your thoughts.

**Comment by marhigh** [comment]
> ok, understand – updated the hover effect accordingly.

Yep that looks good!

**Comment by marcoow** [comment]
@marhigh updated heights look good – added the suggestions to the respective commit 👍

**Pull request merged**